flocker/control/_protocol.py

```diff
@@ -253,7 +313,19 @@ def __init__(self, agent):
         CommandLocator.__init__(self)
         self.agent = agent
 
+    @property
```

Comment:
It seems like a bit of a headache to have a million different `Logger` instances and have to keep track of which log messages go to which ones.  Particularly in this case, where `_AgentLocator` is private.  Forcing the tests to know that this private helper's logger attribute is the one where some log events will go seems like an unfortunately leaky abstraction.

It seems like it kinda makes sense to say "Log events related to this one `IConvergenceAgent` provider you've got will all go to its `logger` attribute" and then make all the things hanging off of that object, including this AMP code, use the object specified by that attribute.  In which case this isn't really ugly, it's just how you have to do things with eliot.

Maybe we should think about whether we like how eliot encourages you to have tons of `Logger` instances all of the place, though.